skipruntime-ts/api/src/internals.ts

```diff
@@ -1,13 +1,34 @@
 const opaque: unique symbol = Symbol.for("Skip.opaque");
 export const sk_frozen: unique symbol = Symbol.for("Skip.frozen");
 
+/**
+ * Type that is tagged to distinguish it from the underlying untagged type.
+ *
+ * The type `Opaque<Value, Tag>` is equivalent to `Value` but is "tagged" with type `Tag` so that it is structurally distinct from `Value`.
+ */
 export type Opaque<Value, Tag> = Value & { [opaque]: Tag };
+
 export type int = number;
+
+/**
+ * Potentially `null` values.
+ *
+ * The type of potentially `null` values of type `T`.
+ */
 export type Nullable<T> = T | null;
 
-export interface Constant {
+/**
+ * Values that are either unmodifiable or managed by the Skip Runtime.
+ *
+ * A `Constant` value is either managed by the Skip Runtime, in which case its modifications are carefully tracked by the reactive computation system, or it is deep-frozen, meaning that it cannot be modified and neither can its sub-objects, recursively.
+ * See `deepFreeze` to make an object `Constant`.
+ *
+ * `Constant` values are important because they can be used in code that will be executed by the reactive computation system without introducing the possibility of producing stale or unreproducible results.
+ */
+export type Constant = {
```

Comment:
To call this change out: I think it is clearer to use `type` over `interface` in cases where user code is not expected to / should not / cannot define its own subtypes.

Reply:
However, the doc generator doesn't do well for `type`s that have methods, so probably the net result is worse. :-(

Reply:
No methods here though, right? Or am I misunderstunding the issue with the doc generator?

Reply:
Ah, yes, here it is ok, I was getting confused with `Context` which can't be implemented by users, but does have methods.